### PR TITLE
Run Cargo fmt and setup CI to verify it

### PR DESCRIPTION
**Pull request by coreyja**
`cargo fmt` can be used to format your code and keeps most Rust code looking relatively consistent which is nice!

This formats the existing code and setups up a CI job to enforce the format in PRs

In VS Code you can configure it to run format on save, so that this happens automatically and you don't really need to think about it at all!

https://stackoverflow.com/a/72282191

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -109,7 +109,7 @@ deny = [
   #"Nokia",
 ]
 # Lint level for licenses considered copyleft
-copyleft = "warn"
```
Comment (coreyja) [inline deny.toml:112]: Oh I snuck in small change here too!

Making this tool help make sure we don't accidentally install copyleft libraries that might force us to open source this code. 

Its 'public' so its more "source available" right now. But I've marked the copyright here as all rights reserved

File: deny.toml [hunk 2]
```diff
@@ -133,6 +133,7 @@ exceptions = [
   # Each entry is the crate and version constraint, and its specific allow
   # list
   #{ allow = ["Zlib"], name = "adler32", version = "*" },
+  { allow = ["MPL-2.0"], name = "webpki-roots", version = "*" },
```
Comment (coreyja) [inline deny.toml:136]: This crate uses the `MPL-2.0` license, and thats considered copy-left but we are allowed to distribute executables with it still so its going to be fine for any of our use cases

File: db/src/lib.rs [hunk 3]
```diff
@@ -40,27 +38,28 @@ pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<U
...
-    )
-    .fetch_optional(pool)
-    .await?;
+pub async fn get_username_by_id(
+    pool: &PgPool,
+    user_id: Uuid,
+) -> color_eyre::Result<Option<String>> {
+    let record = sqlx::query!("SELECT user_name FROM Users WHERE user_id = $1", user_id)
+        .fetch_optional(pool)
+        .await?;
```
Comment (Seif-Mamdouh) [inline db/src/lib.rs:58]: I see it formats the code in a certain way. That's nice!

**Comment by Seif-Mamdouh** [comment]
Looks great! `cargo fmt` is a great library to format the code it makes it look cleaner! 

I read https://jimbobbennett.dev/blogs/cargo-fmt/ which helped me understand the full picture.

**Review by Seif-Mamdouh — DISMISSED** [review]

**Pull request merged**